### ISSUE TITLE
sampling loop time step

### ISSUE THREAD

**Issue by MichaelJacob914** [lucidrains/denoising-diffusion-pytorch#332]
Currently running the code, and after 1000 steps it starts the 'sampling loop time step' but says it will do this 3125 times and is estimated to take about a week. Is this a part of training, and why is it taking so long/will it do this every 1000 steps? Thanks 
<img width="683" alt="Screenshot 2024-07-10 at 4 05 52 PM" src="https://github.com/lucidrains/denoising-diffusion-pytorch/assets/102377228/ceeeca9f-e151-4383-9762-e13a9fbcc0f5">

**Comment by MichaelJacob914** [comment]
It seems as though this was caused by calculate_fid set to true in my trainer statement.

**Comment by Lucien20000118** [comment]
I think it is because of using ddim_sampling. If your `sampling_timesteps < timesteps`, it will make the model use ddim_sample.
If you want to decrease the fid loop numbers, you can set the `trainer.num_fid_samples`  smaller.
The default num_fid_samples is 50000.

**Comment by 4A2008** [comment]
I encountered the same problem, have you figured it out？

**Comment by MichaelJacob914** [comment]
Yup, this part is code to measure the accuracy of the images produced but doesn't help the training, in the code to define the model set calculate_fid = false and it should go away
